components/core/tests/test-clp_s-search.cpp

```diff
@@ -56,6 +67,33 @@ auto get_test_input_local_path() -> std::string {
     return (tests_dir / get_test_input_path_relative_to_tests_dir()).string();
 }
 
+auto create_first_record_match_metadata_query() -> std::shared_ptr<clp_s::search::ast::Expression> {
+    auto zero_literal = clp_s::search::ast::Integral::create_from_int(0);
+    auto one_literal = clp_s::search::ast::Integral::create_from_int(1);
+    auto non_matching_idx = clp_s::search::ast::ColumnDescriptor::create_from_escaped_tokens(
+            {std::string{clp_s::constants::cLogEventIdxName}},
+            std::string{clp_s::constants::cDefaultNamespace}
+    );
+    auto object_subtree_non_matching_idx = non_matching_idx->copy();
```

Comment:
Maybe change idx to column? I personally prefer something like `column`, `column_with_object_subtree_type`, `column_with_metadata_subtree_type`